cloud/openstack/os_server_volume.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/python
+#coding: utf-8 -*-
+
+# Copyright (c) 2014 Hewlett-Packard Development Company, L.P.
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+
+try:
+    import shade
+    from shade import meta
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+
+DOCUMENTATION = '''
+---
+module: os_server_volume
+short_description: Attach/Detach Volumes from OpenStack VM's
+extends_documentation_fragment: openstack
+description:
+   - Attach or Detach volumes from OpenStack VM's
+options:
+   state:
+     description:
+        - Indicate desired state of the resource
+     choices: ['present', 'absent']
+     default: present
+   server:
+     description:
+       - Name or ID of server you want to attach a volume to
+     required: true
+     default: None
+   volume:
+     description:
+      - Name or id of volume you want to attach to a server
+     required: true
+   device:
+     description:
+      - Device you want to attach
+     required: false
+     default: None
+   state:
+     description:
+       - Should the resource be present or absent.
+     choices: [present, absent]
+     default: present
+requirements: ["shade"]
+'''
+
+EXAMPLES = '''
+# Attaches a volume to a compute host
+- name: attach a volume
+  hosts: localhost
+  tasks:
+  - name: attach volume to host
+    os_server_volume:
+      state: present
+      cloud: mordred
+      server: Mysql-server
+      volume: mysql-data
+      device: /dev/vdb
+'''
+
+
+def main():
+    argument_spec = openstack_full_argument_spec(
+        server=dict(required=True),
+        volume=dict(required=True),
+        device=dict(default=None),
```

Comment:
Device should default to auto perhaps, as a convenience, since the data is returned about where it's mounted anyway.